src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,216 @@
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.AddExplicitCast
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddExplicitCast), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ChangeToYield, Before = PredefinedCodeFixProviderNames.GenerateConstructor)]
+    internal partial class AddExplicitCastCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        /// <summary>
+        /// CS0266: Cannot implicitly convert from type 'x' to 'y'. An explicit conversion exists (are you missing a cast?)
+        /// </summary>
+        private const string CS0266 = nameof(CS0266);
+
+        /// <summary>
+        /// CS1503: Argument 1: cannot convert from 'double' to 'int'
+        /// </summary>
+        private const string CS1503 = nameof(CS1503);
+
+        [ImportingConstructor]
+        public AddExplicitCastCodeFixProvider()
+        {
+        }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(CS0266, CS1503); }
+        }
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle; // ?
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var diagnostic = context.Diagnostics.First();
+
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var targetNode = (ExpressionSyntax)TryGetTargetNode(root, diagnostic.Location.SourceSpan);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (CanReplace(semanticModel, root, targetNode, cancellationToken))
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    CSharpFeaturesResources.Add_Explicit_Cast,
+                    c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                    context.Diagnostics);
+            }
+        }
+
+        // Output the current type info of the target node and the conversion type that the target node is going to be casted by
+        private void GetTypeInfo(SemanticModel semanticModel, SyntaxNode root, SyntaxNode targetNode, CancellationToken cancellationToken, out ITypeSymbol nodeType, out ITypeSymbol conversionType)
+        {
+            var nodeInfo = semanticModel.GetTypeInfo(targetNode, cancellationToken);
+            nodeType = nodeInfo.Type;
+            conversionType = nodeInfo.ConvertedType;
+
+            // CS1503 
```

Comment:
i don't undertsand this comment.

Reply:
Oh I think I forget to re-comment it, it is an error code that the following code will fix